### PR TITLE
Add specific error type for "operation already exists" error.

### PR DESCRIPTION
**Pull request by jsafrane**
PersistentVolume controller needs to know why scheduling a new operation has failed - if the operation was already running or some other error happened.

### CODE REVIEW

**Comment by jsafrane** [comment]
@saad-ali, PTAL

File: pkg/util/goroutinemap/goroutinemap.go [hunk 1]
```diff
@@ -74,3 +74,28 @@ func (grm *goRoutineMap) operationComplete(operationName string) {
...
+
+// alreadyExistsError is specific error returned when NewGoRoutine()
+// detects that operation with given name is already running.
+type alreadyExistsError struct {
+	operationName string
+}
+
+var _ error = alreadyExistsError{}
+
+func (err alreadyExistsError) Error() string {
```
Comment (saad-ali) [inline pkg/util/goroutinemap/goroutinemap.go]: nit: This method does not need to be public--it should only be called internally.

File: pkg/util/goroutinemap/goroutinemap.go [hunk 2]
```diff
@@ -74,3 +74,28 @@ func (grm *goRoutineMap) operationComplete(operationName string) {
...
+	return fmt.Sprintf("Failed to create operation with name %q. An operation with that name already exists", err.operationName)
+}
+
+func newAlreadyExistsError(operationName string) error {
+	return alreadyExistsError{operationName}
+}
+
+func IsAlreadyExists(err error) bool {
```
Comment (saad-ali) [inline pkg/util/goroutinemap/goroutinemap.go]: nit: add comment on public method

File: pkg/util/goroutinemap/goroutinemap.go (hunk 1 above)
Comment (jsafrane) [inline pkg/util/goroutinemap/goroutinemap.go]: Not really, it's part of `error` interface

File: pkg/util/goroutinemap/goroutinemap.go (hunk 2 above)
Comment (jsafrane) [inline pkg/util/goroutinemap/goroutinemap.go]: comment added.

File: pkg/util/goroutinemap/goroutinemap.go (hunk 1 above)
Comment (saad-ali) [inline pkg/util/goroutinemap/goroutinemap.go]: Ack

**Comment by jsafrane** [comment]
Rebased

**Comment by jsafrane** [comment]
@k8s-bot test this issue: #22655

**Comment by jsafrane** [comment]
@k8s-bot test this issue: #25673

**Comment by jsafrane** [comment]
@swagiaal, can you please briefly check that I did my rebase right and restore Saad's LGTM?

**Pull request merged**